File: exercism/clojure/gigasecond/src/gigasecond.clj
(ns gigasecond)

(defn from [year month day]  ;; <- arglist goes here
    ;; your code goes here
  (let [next-date (. (java.time.LocalDateTime/of year month day 0 0) plus  (java.time.Duration/ofSeconds 1000000000))]
    [(. next-date getYear) (. next-date getMonthValue) (. next-date getDayOfMonth)]))

File: exercism/java/gigasecond/src/main/java/Gigasecond.java
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    LocalDateTime after;
    public Gigasecond(LocalDate moment) {
        this.after = moment.atTime(0, 0).plusSeconds(1_000_000_000L);
    }

    public Gigasecond(LocalDateTime moment) {
        this.after = moment.plusSeconds(1_000_000_000L);
    }

    public LocalDateTime getDateTime() {
        return this.after;
    }
}
